src/client/options/mod.rs

```diff
@@ -675,40 +675,170 @@ impl Serialize for ClientOptions {
+    ///
+    /// The default value is true.
     pub retry_reads: Option<bool>,
+
+    /// Whether or not the client should retry a write operation if the operation fails.
+    ///
+    /// The default value is true.
     pub retry_writes: Option<bool>,
-    pub socket_timeout: Option<Duration>,
-    pub zlib_compression: Option<i32>,
+
+    /// Specifies whether the Client should directly connect to a single host rather than
+    /// autodiscover all servers in the cluster.
+    ///
+    /// The default value is false.
     pub direct_connection: Option<bool>,
+
+    /// The credential to use for authenticating connections made by this client.
     pub credential: Option<Credential>,
+
+    /// Default database for this client.
+    ///
+    /// By default, no default database is specified.
     pub default_database: Option<String>,
-    max_staleness: Option<Duration>,
-    tls_insecure: Option<bool>,
-    auth_mechanism: Option<AuthMechanism>,
-    auth_source: Option<String>,
-    auth_mechanism_properties: Option<Document>,
-    read_preference: Option<ReadPreference>,
-    read_preference_tags: Option<Vec<TagSet>>,
-    load_balanced: Option<bool>,
+
+    /// Whether or not the client is connecting to a MongoDB cluster through a load balancer.
+    pub load_balanced: Option<bool>,
+
+    /// Amount of time spent attempting to send or receive on a socket before timing out; note that
+    /// this only applies to application operations, not SDAM.
+    pub socket_timeout: Option<Duration>,
+
+    /// Specifies the level of compression when using zlib to compress wire protocol messages; -1
+    /// signifies the default level, 0 signifies no compression, 1 signifies the fastest speed, and
+    /// 9 signifies the best compression.
+    pub zlib_compression: Option<i32>,
+
+    /// The maximum replication lag, in wall clock time, that a secondary can suffer and still be
+    /// eligible for server selection.
+    pub max_staleness: Option<Duration>,
+
+    /// The authentication mechanism method to use for connection to the server.
+    pub auth_mechanism: Option<AuthMechanism>,
+
+    /// The database that connections should authenticate against.
+    pub auth_source: Option<String>,
+
+    /// Additional options provided for authentication (e.g. to enable hostname canonicalization
+    /// for GSSAPI).
+    pub auth_mechanism_properties: Option<Document>,
+
+    /// Default read preference for the client (excluding tags).
+    pub read_preference: Option<ReadPreference>,
+
+    /// Default read preference tags for the client; only valid if the read preference mode is not
+    /// primary.
+    ///
+    /// The order of the tag sets in the read preference is the same as the order they are
+    /// specified in the URI.
+    pub read_preference_tags: Option<Vec<TagSet>>,
+
+    /// Amount of time spent attempting to check out a connection from a server's connection pool
+    /// before timing out.  Not supported by the Rust driver.
```

Comment:
why include this and the next one that are unsupported? what will happen if users try to use them?

Reply:
It seemed useful to include them for domain modeling, i.e. even if the driver doesn't support them customers could use this to parse connection strings and look at those values.  I've moved them to private fields with read-only accessors to make it more clear they shouldn't be set.

Current behavior is the driver ignores those fields.  I'd prefer it if attempting to use them caused an invalid argument error, but that would be a breaking change :(